h/static/scripts/controllers/search-bar-controller.js

```diff
@@ -137,18 +136,48 @@ class SearchBarController extends Controller {
     };
 
     /**
-     * Updates the hidden input field with the consolidated
-     * search terms from the lozenges.
+     * Insert a hidden <input> into the search <form>.
+     *
+     * The value of the hidden <input> is a search string constructed from
+     * the lozenges plus any text currently in the visible <input>.
+     *
+     * The name="q" attribute is moved from the visible <input> on to the
+     * hidden <input> so that when the <form> is submitted it's the value of
+     * the _hidden_ input, not the visible one, that is submitted as the
+     * q parameter.
+     *
      */
-    var updateHiddenInputValue = () => {
-      var newValue = '';
-      Array.from(this._lozengeContainer.querySelectorAll('.js-lozenge__content'), function(loz) {
-        newValue = newValue + loz.textContent + ' ';
+    var insertHiddenInput = () => {
+      var hiddenInput = document.createElement('input');
```

Comment:
I'm inserting the hidden input immediately before submitting the form. @robertknight mentioned that it's more conventional for JavaScript controllers to do their DOM modification at controller initialization time. There are a couple of advantages to doing this at form submit time instead, though:

* No empty hidden input sitting in the DOM not doing anything (there's literally no value text going into it as the user types and things get lozengified, _until_ the moment of form submit)

* You can have a single `insertHiddenInput()` method that does it all, instead of needing one method to insert the hidden input into the DOM at controller init time, and then a second method to set the value of the hidden input at form submit time

* `hiddenInput` can be a local variable of this function (I was able to delete a class-level variable)